homeassistant/components/unifi/device_tracker.py

```diff
@@ -250,16 +212,25 @@ def entity_registry_enabled_default(self):
     async def async_added_to_hass(self):
         """Subscribe to device events."""
         LOGGER.debug("New UniFi device tracker %s (%s)", self.name, self.device.mac)
+        self.device.register_callback(self.async_update_callback)
+        self.listeners.append(
+            async_dispatcher_connect(
+                self.hass, self.controller.signal_reachable, self.async_update_callback
+            )
+        )
 
-    async def async_update(self):
-        """Synchronize state with controller."""
-        await self.controller.request_update()
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect device object when removed."""
+        self.device.remove_callback(self.async_update_callback)
+        for unsub_dispatcher in self.listeners:
+            unsub_dispatcher()
 
```

Comment:
Most of this is a carbon copy from how Ive done it in the deCONZ integration, and there have been not issues around that (so far), since added to hass won't be called until the entity is recreated . If you think it is worth it I can of course do it

Reply:
I don't know if this will ever happen, that this is somehow called twice in a row. We're using both patterns in the code base.

Reply:
Then I'll keep it as it is, if there is no specific path you want to take here